### ISSUE TITLE
impl From<T> for U triggers infinite recursion via .into() — not detected by unconditional_recursion

### ISSUE THREAD

**Issue by HackingRepo** [rust-lang/rust#148742]
Safe Rust code using impl From<E> for String recursively calls .into() on a fresh E {} instance, leading to a stack overflow at runtime. The compiler does not emit a warning or lint, even though the recursion is unconditional and deterministic. This bypasses the compiler recursion check.

Example:
```rust
struct E;
impl E {
    fn new() -> Self {
        E {

        }
    }
}
impl From<E> for String {
    fn from(value: E) -> Self {
        E {}.into()
    }
}

fn main() {    
    let v = String::from(E {});
}


```

When i run this i get
```java
thread 'main' (6844) has overflowed its stack
```

I suggest to Emit a warning when From<T> calls .into() on a fresh T without branching.
